xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-api/src/main/java/org/xwiki/rendering/macro/wikibridge/WikiMacroParameterDescriptor.java

```diff
@@ -37,34 +42,33 @@ public class WikiMacroParameterDescriptor implements ParameterDescriptor
      * 
      * @since 2.1M1
      */
-    private String id;
-
-    /**
-     * Display name of the parameter.
-     * 
-     * @since 2.1M1
-     */
-    private String name;
+    private final String id;
 
     /**
      * Description of the parameter.
      */
-    private String description;
+    private final String description;
 
     /**
      * Boolean indicating if the parameter is mandatory.
      */
-    private boolean mandatory;
+    private final boolean mandatory;
 
     /**
      * Default value of the parameter.
      */
-    private Object defaultValue;
+    private final Object defaultValue;
 
     /**
      * Type of the parameter.
      */
-    private Type parameterType;
+    private final Type parameterType;
+
+    private boolean advanced;
+    private boolean displayHidden;
```

Comment:
Why not simply ``hidden``? "displayHidden" sounds strange / confusing to me.

Reply:
The getter is `isDisplayHidden`, and I'm not inventing it, it's part of the interface there: https://github.com/xwiki/xwiki-rendering/blob/master/xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/macro/descriptor/ParameterDescriptor.java#L116. So I'm just being consistent with the field name. Note that we use `displayHidden` because `hidden` is apparently another bean property, see: https://github.com/xwiki/xwiki-commons/blob/master/xwiki-commons-core/xwiki-commons-properties/src/main/java/org/xwiki/properties/annotation/PropertyHidden.java